packages/store-ui/src/molecules/IconButton/stories/IconButton.stories.tsx

```diff
@@ -0,0 +1,34 @@
+import type { Story, Meta } from '@storybook/react'
+import React from 'react'
+
+import type { Props as IconButtonProps } from '../IconButton'
+import Component from '../IconButton'
+import ShoppingCartIcon from '../../../atoms/Icon/stories/assets/ShoppingCart'
+import mdx from './IconButton.mdx'
+
+const IconButtonTemplate: Story<IconButtonProps> = (props) => (
+  <Component {...props} icon={<ShoppingCartIcon />} />
+)
+
+export const Default = IconButtonTemplate.bind({})
+
+const IconButtonCustomTemplate: Story<IconButtonProps> = (props) => {
+  return (
+    <Component
+      {...props}
+      data-custom-store-icon-button
```

Comment:
I didn't like this custom one. People will already import the component with the `data-store-icon-button`, so why one more data-store-id `data-custom-store-icon-button`?

The current default one is picking the Button's default style, so you can overwrite it with the `data-store-icon-button` and make only one example of this component with the current Custom Icon style.

Reply:
I believe the ideal is not to leave it open for the developer to create new data store IDs. In this case, you are creating the new data-store-id when using the component, not inside the component.

Reply:
> I didn't like this custom one. People will already import the component with the data-store-icon-button, so why one more data-store-id data-custom-store-icon-button?

IMO we are doing components to be flexible and let the developer does customizations.  I did that because I don't want to share the same layout with the default button, inheriting the style from the "data-store-button". So I created a new data attribute for it. I could've added a variation for the data-store-icon-button, but I don't think it's a good idea.

Reply:
> I believe the ideal is not to leave it open for the developer to create new data store IDs. In this case, you are creating the new data-store-id when using the component, not inside the component.

Why not? For me, it doesn't make sense, since we are doing the most flexible component as we can.

Reply:
> I did that because I don't want to share the same layout with the default button, inheriting the style from the "data-store-button".

My point is exactly what you said. I believe we don't want to show a story with an example that inherits the button style and share the same layout, so we'll overwrite everything only once. IMO we don't want to have an example with the base button style, only one showing the icon with one style. 

Also, there is no difference between Default and Custom stories when we are not using a CSS theme, they don't have any differences between their functionalities. That's why I see no reason to have two stories.

Reply:
> Why not? For me, it doesn't make sense, since we are doing the most flexible component as we can.

Idk, I believe it's not needed since we already have a data-store-id for that.

Reply:
Why not data-variant="clean" or something like this? this way we have a standard way of creating variants

Reply:
what do you think about using the current `data-store-icon-button` with a value, something like that:
```css
[data-store-icon-button="story"] {
   css here
}
```
```tsx
   <Component data-store-icon-button="story" />
```

Reply:
The data-attribute already exists on IconButton

Reply:
I'm not sure, I see these `data-store-{component}` as a marker for the component, but the state could be set in a different marker to split the concerns of "entity" and "state". Something similiar to what we do here: https://github.com/vtex/faststore/blob/6d76cecc851b0f67f7a96de7631f2373694d9fd3/packages/store-ui/src/atoms/Price/Price.tsx#L34

The way they do it in there is by adding a different `data-{}`, but I think we ccould do it by using `data-variant="value"`

Reply:
This isn't a state, this is a custom variation of the style.
Take a look at https://reach.tech/menu-button/#menuitem-css-selectors, https://reach.tech/checkbox/#customcheckbox-css-selectors